optaplanner-core/src/test/java/org/optaplanner/core/api/score/buildin/simplelong/SimpleLongScoreHolderTest.java

```diff
@@ -49,11 +50,47 @@ public void addConstraintMatch(boolean constraintMatchEnabled) {
         callOnUpdate(scoreRule3);
         scoreHolder.addConstraintMatch(scoreRule3, -3L); // Overwrite existing
 
-        assertEquals(SimpleLongScore.valueOfUninitialized(0, -1003L), scoreHolder.extractScore(0));
-        assertEquals(SimpleLongScore.valueOfUninitialized(-7, -1003L), scoreHolder.extractScore(-7));
+        assertEquals(SimpleLongScore.ofUninitialized(0, -1003L), scoreHolder.extractScore(0));
+        assertEquals(SimpleLongScore.ofUninitialized(-7, -1003L), scoreHolder.extractScore(-7));
         if (constraintMatchEnabled) {
-            assertEquals(SimpleLongScore.valueOf(-1000L), findConstraintMatchTotal(scoreHolder, "scoreRule1").getScore());
+            assertEquals(SimpleLongScore.of(-1000L), findConstraintMatchTotal(scoreHolder, "scoreRule1").getScore());
         }
     }
 
+    @Test
+    public void rewardPenalizeWithConstraintMatch() {
+        rewardPenalize(true);
+    }
+
+    @Test
+    public void rewardPenalizeWithoutConstraintMatch() {
+        rewardPenalize(false);
+    }
+
+    public void rewardPenalize(boolean constraintMatchEnabled) {
+        SimpleLongScoreHolder scoreHolder = new SimpleLongScoreHolder(constraintMatchEnabled);
+        Rule constraint1 = mockRule("constraint1");
+        scoreHolder.putConstraintWeight(constraint1, SimpleLongScore.of(10L));
+        Rule constraint2 = mockRule("constraint2");
+        scoreHolder.putConstraintWeight(constraint2, SimpleLongScore.of(100L));
+
+        scoreHolder.penalize(mockRuleContext(constraint1));
+        assertEquals(SimpleLongScore.of(-10L), scoreHolder.extractScore(0));
+
+        scoreHolder.penalize(mockRuleContext(constraint2), 2L);
+        assertEquals(SimpleLongScore.of(-210L), scoreHolder.extractScore(0));
+
+        scoreHolder = new SimpleLongScoreHolder(constraintMatchEnabled);
```

Comment:
Doesn't it do that? Line 77 to 82:

```
        scoreHolder.penalize(mockRuleContext(constraint1));
        assertEquals(SimpleScore.of(-10), scoreHolder.extractScore(0));

        scoreHolder.penalize(mockRuleContext(constraint2), 2);
        assertEquals(SimpleScore.of(-210), scoreHolder.extractScore(0));
```

Reply:
I did that originally, but it made the test code less obvious. It would be an ok thing to do, but I don't think it's necessary - I 'll leave it as is.